mount/mount_unix.go

```diff
@@ -85,3 +108,20 @@ func RecursiveUnmount(target string) error {
 	}
 	return nil
 }
+
+// RecursiveUnmount unmounts the target and all mounts underneath, starting
+// with the deepest mount first. The argument does not have to be a mount
+// point itself.
+func RecursiveUnmount(target string) error {
+	// Fast path, works if target is a mount point that can be unmounted.
+	// On Linux, mntDetach flag ensures a recursive unmount.  For other
+	// platforms, if there are submounts, we'll get EBUSY (and fall back
+	// to the slow path). NOTE we do not ignore EINVAL here as target might
+	// not be a mount point itself (but there can be mounts underneath).
+	if err := unix.Unmount(target, mntDetach); err == nil {
+		return nil
+	}
+
+	// Slow path: unmount all mounts inside target one by one.
+	return UnmountAll(target)
```

Comment:
It seems that you need to unmount the target itself here.

Reply:
If I am not mistaken above, this function now works incorrectly on its slow path, and thus our tests are not adequate.

Surely there needs to be a test added for `UnmountAll`.